### DISCUSSION TITLE
Path finding over lora mesh

### DISCUSSION THREAD

**Discussion by attermann** [markqvist/Reticulum#466] [Help & Questions] [answered]
Starting this thread to carry on the discussion about path requests not being re-broadcast over the same interface that they were received on.

This presents a fundamental challenge of not being able to reliably perform path-finding over a LoRa mesh. This severely limits the usefulness of Reticulum over a mesh network since all destinations more than 1 hop away would essentially remain un-routable to any newly joining node.

> markqvist: This is incorrect. Any newly joining node can resolve paths to any destination in the network immediately, there just needs to be an established transport node available nearby.

I assume that by "an established transport node available nearby" you mean as a direct neighbor, right? What if your neighbor is not established either, or only semi-established? This would be the nature of a mesh in the real world (even if nodes are stationary), yes?

**Comment by LinuxinaBit** [comment, marked as answer]
The keys and path to a destination can be requested from any connected Transport Node if your node does not have an announce from that destination.
No node has a map of the complete network it is connected to, instead simply storing the next hop to the destination for every endpoint.

> I assume that by "an established transport node available nearby" you mean as a direct neighbor, right?

No. A transport is just any node that routes traffic and stores announces. By “established” Mark means that is has been on long enough to receive the announce for the destination you are trying to reach. This can actually be any transport node on the entire network, just as long as you’re able to connect to it.

Please read this for more information: https://reticulum.network/manual/understanding.html
Reticulum is very confusing until you fully understand how the Announce mechanism works. I am working on a video explainer to help with this, but currently your best option is to read that chapter of the manual and remember that Reticulum is likely very different from a lot of other mesh networking projects you’ve seen before.
If you are curious as to why not *every* node is a transport node (why not every node routes traffic), read this after finishing that part of the manual: #422

**Reply by markqvist** [reply]
Just to add a small comment to this really good answer, design decisions like this were made intentionally to encourage good network design.

If we want functional and reliable networking in very challenging situations (like ultra-low bandwidth and unreliable mediums), it is necessary to have *both* a highly efficient stack *and* make good decisions on how to build and deploy the networks.

Reticulum actively tries to encourage making sensible network design decisions, and it also seeks to _not promise things that are physically impossible_ within the envelope of currently available radio technology, or lead users down untenable paths.
